aws_topology/stackstate_checks/aws_topology/resources/api_gateway.py

```diff
@@ -133,13 +133,11 @@ def process_all(self, filter=None):
                                     method_integration_uri.rfind('arn'):method_integration_uri.find('/invocations')
                                 ]
                             elif re.match("arn:aws:apigateway:.+:sqs:path/.+", method_integration_uri):
-                                #  TODO cross region fails
-                                queue_name = method_integration_uri.rsplit('/', 1)[-1]
                                 queue_arn = arn(
-                                    'sqs',
-                                    self.location_info['Location']['AwsRegion'],
-                                    self.location_info['Location']['AwsAccount'],
-                                    queue_name
+                                    resource='sqs',
+                                    region=method_integration_uri.split(':', 4)[3],
```

Comment:
You have access to the `location_info` right? That looks a lot safer than splitting on the uri. If the location is not an option, just check these splittings before looking up the index they might fail pretty easily.

Reply:
This is very much on purpose, because if we use `location_info` we assume that the target resource is in the same AWS account and region. This breaks when the resource exists in a different account or region